src/ir/types.py

```diff
@@ -107,6 +124,19 @@ def get_supertypes(self):
                     stack.append(supertype)
         return visited
 
+    def substitute_type_args(self, type_map,
+                             cond=lambda t: t.has_type_variables()):
```

Comment:
Why do we have this method inside the generic class `Type`?. I mean a generic type has nothing to do with type arguments.